jobsdb/jobsdb.go

```diff
@@ -2556,6 +2577,11 @@ func (jd *HandleT) addNewDSLoop(ctx context.Context) {
 							if err = setReadonlyDsInTx(tx, latestDS); err != nil {
 								return fmt.Errorf("error making dataset read only: %w", err)
 							}
+						} else {
+							// maybe another node added a new DS that we need to make visible to us
+							if err := jd.refreshDSList(ctx); err != nil {
+								return fmt.Errorf("refreshDSList: %w", err)
+							}
```

Comment:
Why can't we use the fresh `dsList` we fetch above in line.2549 to set `jd.dsList` here. Would need the `asyncLock` here as well.

Is it because `migration` might change how the tables look..?

Reply:
Since the last dataset and the one that could be potentially created is(are) of interest, I think something like what you've done in `doRefreshDSList` can be done(comparing the last datasets in both arrays).

As for the rangeList, it's refreshed once the transaction ends anyway, so why here again?